portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx

```diff
@@ -200,6 +204,21 @@ function getActivatedKeyListFromState<KeyType>(
     .map((authenticator) => authenticator.key);
 }
 
+function updateScreenStateField<
+  K extends keyof AuthenticationAuthenticatorScreenState
+>(
+  setState: React.Dispatch<
+    React.SetStateAction<AuthenticationAuthenticatorScreenState>
+  >,
+  field: K,
+  action: React.SetStateAction<AuthenticationAuthenticatorScreenState[K]>
+) {
+  setState((prev) => ({
+    ...prev,
+    [field]: typeof action === "function" ? action(prev[field]) : action,
+  }));
+}
+
```

Comment:
This function seems a bit redundant. Why don't we just use `setState` inline?

Reply:
Let's change to using `setState` inline. I abstract this to a function as it seems repetitive